File: 백준/Silver/1764. 듣보잡/듣보잡.py
import sys

N, M = map(int, sys.stdin.readline().strip().split(" "))
names = {}
for _ in range(N):
    name = sys.stdin.readline().strip()
    names[name] = 1


answer = {}
for _ in range(M):
    name = sys.stdin.readline().strip()
    if name in names.keys():
        answer[name]=1

print(len(answer))
answers = list(answer.keys())
answers.sort()
for name in answers:
    print(name)

File: 백준/Silver/1764. 듣보잡/듣보잡.go
package main

import (
	"os"
	"bufio"
	"fmt"
	"sort"
)

func main(){
	reader := bufio.NewReader(os.Stdin)
	writer := bufio.NewWriter(os.Stdout)
	defer writer.Flush()
	var arr [2]int
	fmt.Fscanln(reader, &arr[0], &arr[1])

	names := make(map[string] int)

	var name string
	count := 0

	for i:=0; i< arr[0]; i++ {
		fmt.Fscanln(reader, &name);
		names[name]++
	}

	var result []string
	for i:=0; i< arr[1]; i++ {
		fmt.Fscanln(reader, &name);
		names[name]++

		if names[name] >= 2 {
			count++
			result = append(result, name)
		}
	}

	fmt.Fprintln(writer, count)

	sort.Strings(result)
	
	for i := 0; i< count; i++ {
		fmt.Fprintln(writer, result[i])
	}
}